server/controllers/request.js

```diff
@@ -0,0 +1,73 @@
+const Request = require("../models/Request");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /request
+// @desc create a new pet sitting request
+// @access Private
+exports.makeRequest = asyncHandler(async (req, res, next) => {
+  const body = req.body;
+  const sitter = await User.findById(body.sitterId);
+  if (!sitter) {
+    res.status(404);
+    throw new Error("Sitter doesn't exist");
+  }
+  const newRequest = await Request.create({
+    ownerId: req.user.id,
+    sitterId: body.sitterId,
+    duration: {
+      start: new Date(body.duration.start),
+      end: new Date(body.duration.end),
+    },
+  });
+  if (newRequest) {
+    res.status(200).json({
+      success: {
+        request: newRequest,
+      },
+    });
+  } else {
+    res.status(500);
+    throw new Error("Error creating request");
+  }
+});
+
+// @route GET /request
+// @desc Get requests for a logged in user
+// @access Private
+exports.getRequest = asyncHandler(async (req, res, next) => {
+  const requests = await Request.find({ sitterId: req.user.id });
+  res.status(200).json({
+    success: {
+      requests: requests,
+    },
+  });
+});
+
+// @route PATCH /request
+// @desc Update a user's request
+// @access Private
+exports.editRequest = asyncHandler(async (req, res, next) => {
+  const body = req.body;
+  const verifyUser = await Request.findOne({ _id: body.id });
+
+  if (req.user.id != (verifyUser.ownerId || verifyUser.sitterId)) {
```

Comment:
Not sure if this statement will give the intended result. Think you could say
`if (req.user.id !== verifyUser.ownerId && req.user.id !== verifyUser.sitterId))`

Reply:
Seems ok: If the requester is not the owner or sitter of this request, then they are unauthorized.